isis/src/base/objs/PvlObject/PvlObject.cpp

```diff
@@ -63,27 +63,14 @@ namespace Isis {
 
     for(auto it = jsonobj.begin(); it != jsonobj.end(); it++) {
         PvlKeyword keyword;
+        keyword.setName(QString::fromStdString(it.key()));
         if (it.value().is_array()) {
-          keyword.setName(QString::fromStdString(it.key()));
-          for(auto ar = it.value().begin(); ar!=it.value().end();ar++) {
-            keyword += QString::number(ar->get<double>(), 'g', 16);
+          for(auto ar = it.value().begin(); ar!=it.value().end(); ar++) {
+            keyword.addJsonValue(*ar);
           }
         }
-        else if(it.value().is_number()) {
-          keyword.setName(QString::fromStdString(it.key()));
-          keyword.setValue(QString::number(it->get<double>()));
-        }
-        else if(it.value().is_boolean()) {
-          keyword.setName(QString::fromStdString(it.key()));
-          keyword.setValue(QString(it->get<bool>() ? "true" : "false"));
-        }
-        else if(it.value().is_null()) {
-          keyword.setName(QString::fromStdString(it.key()));
-          keyword.setValue(QString("Null"));
-        }
         else {
-          keyword.setName(QString::fromStdString(it.key()));
-          keyword.setValue(QString::fromStdString(it.value()));
+          keyword.setJsonValue(*it);
         }
         addKeyword(keyword);
```

Comment:
@jessemapel  When you refer to replacing the constructor is this what you are referring to?

Reply:
It is, I somehow smashed this together with the PvlKeyword functions and thought this was a separate function.